src/Enigma2.cpp

```diff
@@ -347,20 +392,28 @@ PVR_ERROR Enigma2::GetEPGForChannel(ADDON_HANDLE handle, const PVR_CHANNEL &chan
   if (m_epg.IsInitialEpgCompleted() && m_settings.GetEPGDelayPerChannelDelay() != 0)
     Sleep(m_settings.GetEPGDelayPerChannelDelay());
 
-  if (!m_channels.IsValid(channel.iUniqueId))
+  //Have a lock while getting the channel. Then we don't have to worry about a disconnection while retrieving the EPG data.
+  std::shared_ptr<Channel> myChannel = nullptr;
```

Comment:
No need to assign nullptr, will be done automatically by std::shared_ptr ctor (and this time I'm right, remember my brain fart with std::make_shared? :-)

Reply:
Done